spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDistributionAndOrderingUtils.scala

```diff
@@ -57,10 +57,28 @@ object ExtendedDistributionAndOrderingUtils {
         } else {
           conf.numShufflePartitions
         }
-        // the conversion to catalyst expressions above produces SortOrder expressions
-        // for OrderedDistribution and generic expressions for ClusteredDistribution
-        // this allows RepartitionByExpression to pick either range or hash partitioning
-        RepartitionByExpression(ArraySeq.unsafeWrapArray(distribution), query, finalNumPartitions)
+
+        val isHashDistributionMode = write.requiredDistribution match {
+          case _ : ClusteredDistribution => true
+          case _ => false
+        }
+
+        val strictDistributionMode = conf.getConfString(SparkSQLProperties.STRICT_TABLE_DISTRIBUTION,
+          SparkSQLProperties.STRICT_TABLE_DISTRIBUTION_DEFAULT)
+
+        if (strictDistributionMode.equals("false") && isHashDistributionMode) {
```

Comment:
`strictDistributionMode` could convert to a boolean easily with `.toBoolean`

Reply:
Thanks for suggestion @ConeyLiu !